ui/apps/platform/cypress/integration/vulnerabilities/workloadCves/imageCveSingle.test.js

```diff
@@ -0,0 +1,173 @@
+        cy.get(selectors.resourceMenuItem('Cluster'));
+        cy.get(selectors.resourceMenuItem('Namespace'));
+        cy.get(selectors.resourceDropdown).click();
+
+        // The Images toggle button should be the default and the count of the images in the button
+        // should match the count of the images in the "Affected images" card
+        cy.get('button[aria-pressed="true"]:contains("Image")').then((button) => {
+            const buttonImageCount = button.text().replace(/ Image.*/, '');
+
+            cy.get(
+                `${selectors.summaryCard('Affected images')} *:contains("images affected")`
+            ).then((affectedImages) => {
+                const affectedImageCount = affectedImages
+                    .text()
+                    .replace(/\/\d* images affected/, '');
+
+                expect(buttonImageCount).to.equal(affectedImageCount);
+            });
+        });
+    });
+
+    it('should correctly handle local filters on the images tab', () => {
+        visitWorkloadCveOverview();
+        cy.get('tbody tr td a:contains("CVE")').first().click();
+
+        cy.get('table tbody tr:nth-of-type(1) td[data-label="CVE severity"]').then(
+            ([$severity]) => {
+                // Extract the severity from the first row of the table. These are values
+                // that we know should exist in the table when no filters are applied.
+                const severity = $severity.innerText;
+
+                // Apply any filter that _doesn't_ match the severity value extracted from the first row to ensure all data
+                // matching that severity is filtered out from the table. e.g. If the first row has a severity of "Critical",
+                // applying a filter of "Important" will ensure that no data in the table will match a severity of "Critical".
+                cy.get(selectors.severityDropdown).click();
+                cy.get(`${selectors.severityMenuItems} label:not(:contains("${severity}"))`)
+                    .first()
+                    .click();
+                cy.get(selectors.severityDropdown).click();
+
+                // With the negative filters applied, no data in the table should match
+                cy.get(
+                    `table tbody tr td[data-label="CVE severity"]:contains("${severity}")`
+                ).should('not.exist');
+
+                // Clear the filters
+                cy.get(selectors.clearFiltersButton).click();
+
+                // Apply the filter that _does_ match the value extracted from the first row to ensure all data
+                // in the table only contains that severity value. Since this value was pulled from the first row
+                // we know that there will be at least one entry in the table.
+                cy.get(selectors.severityDropdown).click();
+                cy.get(selectors.severityMenuItem(severity)).click();
+                cy.get(selectors.severityDropdown).click();
```

Comment:
If there are enough occurrences of an interaction, a semantic helper function can encapsulate repeated references to same selector. That is,
* test file imports more functions and fewer selectors
* helper file can encapsulate some of the selectors

Reply:
I found a couple more good candidates for extraction, so I pushed an update with those as well.